### ISSUE TITLE
provide method to remap alleles (eg for VCF output)

### ISSUE THREAD

**Issue by petrelharp** [tskit-dev/tskit#1341]
Sometimes we'd like to be able to change what the ancestral/derived states are, for instance, for doing VCF output of a SLiM-produced tree sequence. In #1081 we decided that the way to do this is to provide a method that remaps alleles. It's pretty easy to do this by hand now, so I'm not sure at what level of generality the API should work. In #1081 I proposed [having the method take a callable that takes in a Variant and return a list of new alleles](https://github.com/tskit-dev/tskit/pull/1081/files#diff-ecbb802115b51a593926a3015f4ff2ec444fa4dc4915af74f4959abc60c84df8R4747). This is on the one hand probably too general - we should be trying to make the common use case of converting things to nucleotides very easy - and not completely general - since distinct mutations can have the same derived allele, and this wouldn't let us remap those to different alleles.

For the SLiM-to-VCF problem, I think we want to do something like this, to get unique alleles for each mutation:
```
import tskit, msprime
import random

ts = msprime.sim_ancestry(10, recombination_rate=0.01, sequence_length=100)
ts = msprime.sim_mutations(ts, rate=0.01, model=msprime.SLiMMutationModel(type=0))
t = ts.tables
t.sites.clear()
t.mutations.clear()

def make_alleles(n):
    nucs = ['A', 'C', 'G', 'T']
    alleles = nucs.copy()
    random.shuffle(alleles)
    while len(alleles) < n:
        new = [a + b for a in alleles for b in nucs]
        random.shuffle(new)
        alleles.extend(new)
    return alleles[:n]

for s in ts.sites():
    alleles = make_alleles(len(s.mutations) + 1)
    t.sites.append(s.replace(ancestral_state=alleles[0]))
    for j, m in enumerate(s.mutations):
        t.mutations.append(m.replace(derived_state=alleles[j+1]))

mts = t.tree_sequence()
for h in mts.haplotypes():
    print(h)
```

**Comment by jeromekelleher** [comment]
Yep, LGTM. Although, I think it would be best to put this into PySLIM: add a function called ``ts = pyslim.convert_nucleotides(ts)``. Then when you want to make a VCF from a SLiM tree sequence you call this pyslim function on it. That way, pyslim can check the metadata of the mutations to see if there's already a nucloetide associated.

**Comment by petrelharp** [comment]
That makes sense. Any other use cases for remapping alleles?

**Comment by jeromekelleher** [comment]
Not that I can think of...
